tests/test_linter.py

```diff
@@ -1,8 +1,298 @@
-from crs_linter.linter import Check
+from crs_linter.linter import Check, parse_config
 
 
-def test_check(data, txvars):
-    c = Check(data, txvars)
+def test_parser():
+    t = 'SecRule REQUEST_HEADERS:User-Agent "@rx ^Mozilla" "id:1,phase:1,log,status:403"'
+    p = parse_config(t)
+
+    assert p is not None
+
+def test_check_ignore_proper_case():
+    t = 'SecRule REQUEST_HEADERS:User-Agent "@rx ^Mozilla" "id:1,phase:1,log,status:403"'
+    p = parse_config(t)
+    c = Check(p)
     c.check_ignore_case()
 
     assert len(c.caseerror) == 0
+
+def test_check_ignore_case_fail_invalid_action_case():
+    """ Two actions are in the wrong case. """
+    t = 'SecRule REQUEST_HEADERS:User-Agent "@rx ^Mozilla" "id:1,phase:1,LOG,NoLOg,status:403"'
+    p = parse_config(t)
+    c = Check(p)
+    c.check_ignore_case()
+
+    assert len(c.caseerror) == 2
+
+def test_check_action_order():
+    """ Test that the actions are in the correct order. """
+    t = 'SecRule REQUEST_HEADERS:User-Agent "@rx ^Mozilla" "id:1,phase:1,nolog"'
+    p = parse_config(t)
+    c = Check(p)
+    c.check_action_order()
+
+    assert len(c.orderacts) == 0
+
+def test_check_action_fail_wrong_order():
+    """ Test if the action is in the wrong order. status should go before log """
+    t = 'SecRule REQUEST_HEADERS:User-Agent "@rx ^Mozilla" "id:1,phase:1,log,status:403"'
+    p = parse_config(t)
+    c = Check(p)
+    c.check_action_order()
+
+    assert len(c.orderacts) == 1
+
+def test_check_ctl_auditctl_log_parts():
+    """ Test that there is no ctl:auditLogParts action in any rules"""
+    t = 'SecRule REQUEST_HEADERS:User-Agent "@rx ^Mozilla" "id:1,phase:1,log,status:403"'
+    p = parse_config(t)
+    c = Check(p)
+    c.check_ctl_audit_log()
+
+    assert len(c.auditlogparts) == 0
+
+
+def test_check_wrong_ctl_audit_log_parts():
+    t = 'SecRule REQUEST_HEADERS:User-Agent "@rx ^Pizza" "id:1,phase:1,log,ctl:auditLogParts=+E"'
+    p = parse_config(t)
+    c = Check(p)
+    c.check_ctl_audit_log()
+
+    assert len(c.auditlogparts) == 1
+
+
+def test_check_tx_variable():
+    """ Test that variables are defined in the transaction """
+    t = """SecRule &TX:blocking_paranoia_level "@eq 0" \
+    "id:901120,\
+    phase:1,\
+    pass,\
+    nolog,\
+    ver:'OWASP_CRS/4.0.0-rc1',\
+    setvar:'tx.blocking_paranoia_level=1'"
+
+SecRule &TX:detection_paranoia_level "@eq 0" \
+    "id:901125,\
+    phase:1,\
+    pass,\
+    nolog,\
+    ver:'OWASP_CRS/4.0.0-rc1',\
+    setvar:'tx.detection_paranoia_level=%{TX.blocking_paranoia_level}'"
+    """
+    p = parse_config(t)
+    c = Check(p)
```

Comment:
Before you call `c.check_tx_variable()` you should call `c.collect_tx_variable()`. Otherwise the `globalvars` won't be filled.